File: java/problema1/main/src/Main.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.*;

public class Main {
    public static void main(String[] args) {
        String textoSucio;
        String textoLimpio;
        Map<String, Integer> conteoPalabras = new HashMap<String, Integer>();
        List<Map.Entry<String, Integer>> conteoPalabrasSort;

        try {
            // Lectura del archivo
            FileReader fr = new FileReader("texto_prueba.txt");
            BufferedReader archivo = new BufferedReader(fr);

            textoSucio = archivo.readLine();

            // Se limpian los signos de puntuación para hacer el conteo de palabras
            // mediante una expresión regular
            textoLimpio = textoSucio
                    .replaceAll("[!\"#$%&'()*+,-./:;<=>?@\\[\\]^_`{|}~]", "")
                    .toLowerCase();

            for (String palabra: textoLimpio.split(" ")) {
                if (conteoPalabras.containsKey(palabra)) {
                    conteoPalabras.put(
                            palabra,
                            conteoPalabras.get(palabra) + 1
                    );
                } else {
                    conteoPalabras.put(palabra, 1);
                }
            }

            // Diccionario que almacenará la cantidad de veces que se repite cada palabra
            conteoPalabrasSort = new ArrayList<>(conteoPalabras.entrySet());

            // Ordenamos el diccionario por la cantidad de veces que aparece cada palabra
            Collections.sort(conteoPalabrasSort, (entrada1, entrada2) -> entrada2.getValue().compareTo(entrada1.getValue()));

            // Imprimir el resultado
            for (int i = 0; i < 10; i++) {
                String palabra = conteoPalabrasSort.get(i).getKey();
                Integer numero = conteoPalabrasSort.get(i).getValue();

                System.out.println(i + 1 + ": " + palabra + " - " + numero + " veces");
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: python/problema1/main.py
import string
import time

# Lectura del archivo
archivo = open("texto_prueba.txt", "r")
texto_sucio = archivo.read()
archivo.close()

start_time = time.time()
# string.punctuation contiene todos los signos de puntuación
texto_limpio = ''.join([c for c in texto_sucio if c not in string.punctuation]).lower()

# Diccionario que almacenará la cantidad de veces que aparece cada palabra
conteo_palabras = {}

# Separar el texto en palabras y contar cuántas veces aparece cada una
for palabra in texto_limpio.split(" "):
    if palabra in conteo_palabras:
        conteo_palabras[palabra] += 1
    else:
        conteo_palabras[palabra] = 1

# Ordenamos el diccionario por la cantidad de veces que aparece cada palabra
conteo_palabras_sort = list(
    sorted(conteo_palabras.items(), key=lambda item: item[1], reverse=True)
)
end_time = time.time()

print(f"Tiempo de ejecución: {end_time - start_time} segundos")

# Imprimir el resultado
print("Conteo de palabras")
for i in range(10):
    print(f"{i + 1}: {conteo_palabras_sort[i][0]} - {conteo_palabras_sort[i][1]} veces")
